client/src/pages/room.jsx

```diff
@@ -0,0 +1,252 @@
+import { useContext, useEffect, useState } from 'react';
+import { Context } from '../context/context.jsx';
+import { Formik, Form, Field } from 'formik';
+import { userService } from '../services/userService.js';
+import { useNavigate, useParams } from 'react-router-dom';
+import { ChatBlock } from './ChatsBlock.jsx';
+
+function messageValidation(newMessage) {
+  if (newMessage.trim().length > 300) {
+    return 'Message is to long';
+  }
+}
+
+const socket = new WebSocket('ws://localhost:7070');
+
+export const Room = () => {
+  const {
+    currentRoom,
+    setCurrentRoom,
+    setUser,
+    user,
+    setAllChats,
+    setAnyError,
+  } = useContext(Context);
+
+  const [error, setError] = useState(null);
+
+  const navigate = useNavigate();
+
+  const { roomId } = useParams();
+
+  socket.onmessage = ({ data }) => {
+    console.log('Message from server', JSON.parse(data));
+    setCurrentRoom(JSON.parse(data));
+  };
+
+  useEffect(() => {
+    const storedUser = localStorage.getItem('nickname');
+    setUser(storedUser);
```

Comment:
Consider checking if `storedUser` is not null before calling `setUser(storedUser)`. This will prevent setting a null value to the user state if no nickname is stored in localStorage.